addons/point_of_sale/models/pos_order.py

```diff
@@ -348,10 +352,10 @@ def _default_pricelist(self):
     company_id = fields.Many2one('res.company', string='Company', required=True, readonly=True, default=lambda self: self.env.user.company_id)
     date_order = fields.Datetime(string='Order Date', readonly=True, index=True, default=fields.Datetime.now)
     user_id = fields.Many2one('res.users', string='Salesman', help="Person who uses the cash register. It can be a reliever, a student or an interim employee.", default=lambda self: self.env.uid)
-    amount_tax = fields.Float(compute='_compute_amount_all', string='Taxes', digits=0)
-    amount_total = fields.Float(compute='_compute_amount_all', string='Total', digits=0)
-    amount_paid = fields.Float(compute='_compute_amount_all', string='Paid', states={'draft': [('readonly', False)]}, readonly=True, digits=0)
-    amount_return = fields.Float(compute='_compute_amount_all', string='Returned', digits=0)
```

Comment:
The problem reported at #15727 was that the amounts were not exactly computed the same way in JS and python. It is not the first time we have these and it is a risk to have rounding errors, not marking the orders as paid because there is a 1 cent difference between the payment and total.
Computing the amounts at only one place is the best way to prevent similar errors.
To still allow to make it in the backend, I turned the compute methods into onchange but it is a choice if we want to discourage the order creation from the backend.